region/lowernorfair/west/Screw Attack Room.json

```diff
@@ -1094,43 +1744,122 @@
     {
       "id": 45,
       "link": [3, 4],
-      "name": "Temporary Blue Descent and Shinespark Escape Top Door Part 1",
+      "name": "Come In Shinecharging, End Shinecharged (Temporary Blue)",
       "entranceCondition": {
         "comeInShinecharging": {
           "length": 3,
           "openEnd": 1
         }
       },
       "requires": [
-        {"notable": "Descent and Shinespark Escape"},
         "canTemporaryBlue",
         "canShinechargeMovementTricky",
-        {"heatFrames": 130}
+        {"heatFrames": 155},
+        {"shineChargeFrames": 140}
       ],
-      "clearsObstacles": ["A", "B", "C"],
-      "flashSuitChecked": true,
-      "note": "Simultaneously store a shinespark and break through the bomb blocks down to the item location.",
-      "devNote": "canShinechargeMovementTricky is to represent the difficulty of activating the shinecharge in a precise place near the edge."
+      "endsWithShineCharge": true,
+      "clearsObstacles": ["A", "B"],
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": []
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ],
+      "note": [
+        "Press down precisely to gain the shinecharge while sliding off the ledge.",
+        "Maintain the temporary blue state to break through the blocks at the bottom of the room.",
+        "Reach the item while still having a shinecharge."
+      ]
     },
     {
       "id": 46,
       "link": [3, 4],
-      "name": "Screw Descent and Shinespark Escape Top Door Part 1",
+      "name": "Come In Shinecharging, End Shinecharged (Full Runway, Screw Attack)",
       "entranceCondition": {
         "comeInShinecharging": {
           "length": 12,
           "openEnd": 0
         }
       },
       "requires": [
-        {"notable": "Descent and Shinespark Escape"},
         "ScrewAttack",
-        "canShinechargeMovement",
-        {"heatFrames": 170}
+        "canShinechargeMovementTricky",
+        {"heatFrames": 205},
+        {"shineChargeFrames": 155},
+        {"or": [
+          "canMoonfall",
+          {"and": [
+            {"heatFrames": 5},
+            {"shineChargeFrames": 5}
+          ]}
+        ]}
+      ],
+      "endsWithShineCharge": true,
+      "clearsObstacles": ["B"],
```

Comment:
Makes sense to not count A as broken

Reply:
Yeah, I was thinking about how the doorframe CWJ wouldn't work with only the left blocks broken. We could make a separate obstacle for left vs. right block, but I didn't see a scenario where it would matter